crates/stackable-operator/src/product_config_utils.rs

```diff
@@ -1085,7 +1098,7 @@ mod tests {
                     ),
                 PropertyNameKind::Cli =>
                     collection!(
-                        "cli".to_string() => Some(GROUP_CLI.to_string()),
+                        "cli".to_string() => Some("cli".to_string()),
```

Comment:
question: Do we have tests in place which assert that overrides are now applied in the (new) order we expect? If not, we should add some cases.

Reply:
We are currently adding integration tests as part of https://github.com/stackabletech/issues/issues/548 to the operators. Is that what you had in mind or do you mean additional unit tests other than the one adapted here?

Reply:
Ah, we are adding integration tests - that's also good.

I thought we would also add unit tests, to make sure the code _itself_ does the right thing, independent from what the operator does. For example, build a complete chain of inheritance (role config < role override < role group config < role group override) (See https://github.com/stackabletech/decisions/issues/22) and assert that values are overwritten at the right level.

Reply:
Added some test cases in https://github.com/stackabletech/operator-rs/pull/841/commits/ffd77f80aa434f86f58707486db4b25cc8b69568 :+1: